api/clients/codecs/blob_codec.go

```diff
@@ -1,48 +1,73 @@
 package codecs
 
 import (
+	"bytes"
+	"encoding/binary"
 	"fmt"
+
+	"github.com/Layr-Labs/eigenda/encoding/utils/codec"
 )
 
 type BlobEncodingVersion byte
 
 const (
-	// This minimal blob encoding contains a 32 byte header = [0x00, version byte, uint32 len of data, 0x00, 0x00,...]
+	// DefaultBlobEncoding entails a 32 byte header = [0x00, version byte, uint32 len of data, 0x00, 0x00,...]
 	// followed by the encoded data [0x00, 31 bytes of data, 0x00, 31 bytes of data,...]
 	DefaultBlobEncoding BlobEncodingVersion = 0x0
 )
 
-type BlobCodec interface {
-	DecodeBlob(encodedData []byte) ([]byte, error)
-	EncodeBlob(rawData []byte) ([]byte, error)
-}
+// EncodePayload accepts an arbitrary payload byte array, and encodes it.
```

Comment:
Let's be super precise in what "encode" means. What's the length of the encoded thing? Is it multiple of 32, or can the last element not be? etc